src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,174 @@
+                addTask(splitData[2], LocalDate.parse(splitData[3]),"deadline", Boolean.parseBoolean(splitData[1]));
+                break;
+            case("E"):
+                addTask(splitData[2], LocalDate.parse(splitData[3]), "event", Boolean.parseBoolean(splitData[1]));
+            }
+        }
+    }
+
+    /**
+     * Returns the array of tasks.
+     *
+     * @return Array of tasks.
+     */
+    public ArrayList<Task> getList() {
+        return this.list;
+    }
+
+    /**
+     * Adds a task the list and saves to the file.
+     *
+     * @param description The description of the task.
+     */
+    public String addTask(String description) {
+        Task task = new ToDo(description);
+        this.list.add(task);
+        ui.writeOutput("Task added successfully: \n" + task + "\nNumber of tasks in list: " + list.size());
+        return task.saveString();
+    }
+
+    /**
+     * Adds a task the list without saving to file.
+     *
+     * @param description The description of the task.
+     * @param isDone Indicates if the task is done.
+     */
+    public void addTask(String description, boolean isDone) {
+        Task task = new ToDo(description, isDone);
+        this.list.add(task);
+    }
+
+    /**
+     * Adds a task the list and saves to the file.
+     * Task added have a date/time attached.
+     *
+     * @param description The description of the task.
+     * @param date The date/time attached to the task.
+     * @param type Indicates if the task is a Deadline or Event.
+     */
+    public String addTask(String description, LocalDate date, String type) {
+        Task task;
+        if (type.equals("deadline")) {
+            task = new Deadline(description, date);
+        } else {
+            task = new Event(description, date);
+        }
+        this.list.add(task);
+        ui.writeOutput("Task added successfully: \n" + task + "\nNumber of tasks in list: " + list.size());
+        return task.saveString();
+    }
+
+    /**
+     * Adds a task the list without saving to file.
+     * Task added have a date/time attached.
+     *
+     * @param description The description of the task.
+     * @param dateTime The date associated with the task.
+     * @param type The type of task.
+     * @param isDone Indicates if the task is done.
+     */
+    public void addTask(String description, LocalDate dateTime, String type, boolean isDone) {
+        Task task;
+        if (type.equals("deadline")) {
+            task = new Deadline(description, dateTime, isDone);
+        } else {
+            task = new Event(description, dateTime, isDone);
+        }
+        this.list.add(task);
+    }
+
+    /**
+     * Changes the status for the task indicated by the user.
+     *
+     * @param taskNumber The index(plus 1) of the task to be marked as done.
+     * @return The string representation of the task after it is marked as done.
+     */
+    public String changeTaskStatus(int taskNumber) throws DukeException {
```

Comment:
Do you think it would be better to extract out the checking of the index for both the methods changeTaskStatus and deleteTask into a separate method instead?